python/lsst/daf/butler/registry/interfaces/_database.py

```diff
@@ -831,7 +950,10 @@ def ensureTableExists(self, name: str, spec: ddl.TableSpec) -> sqlalchemy.schema
 
         Subclasses may override this method, but usually should not need to.
         """
-        assert not self._connection.in_transaction(), "Table creation interrupts transactions."
+        # TODO: if _engine is used to make a table then it uses separate
+        # connection and should not interfere with current transaction
+        assert self._session_connection is None or not self._session_connection.in_transaction(), \
+            "Table creation interrupts transactions."
```

Comment:
FWIW, it may be that this already wasn't a problem; it was only with Oracle where I was certain that DDL statements would break transactions (and it may be that with Oracle, using a separate connection for DDL doesn't save you; I'm not sure).  I left this assert in after we dropped Oracle because it seemed like a good idea to leave room for future databases (maybe even Oracle again) where that might be a problem, and without the check it would be very easy to write important code that would be hard to use with such a database.

Reply:
I beleive that in Postgres CREATE TABLE is transactional, but I know at least one other RDBMS (mysql) that does not support it, in mysql CREATE TABLE implicitly commits transaction. In general DDL statements are not required to be transaction-safe so it is probably worth keeping this assert just in case.